doc/help/CHECK.md

```diff
@@ -0,0 +1,83 @@
+kdb-check(1) -- Perform internal checks
+=======================================
+
+## DESCRIPTION
+This command is used to perform checks on the key database or an elektra plugin.  
+
+## USAGE
+`kdb check <plugin>`  
+
+Where `plugin` is the plugin that a user wants to check.  
+Use `-c` to pass options to that plugin.  
+If no `plugin` argument is provided a check will be performed on the key database itself.  
+The user can also use this tool to perform write tests by passing the `-f` option (please note that this can result in configuration files being changed!).  
+Special values are [returned](#RETURNED VALUES) to represent the outcome of a check.  
+
+## RETURNED VALUES
+Their are two different types of checks, a check on a plugin (by specifying the name of a plugin as an argument) or a check on the key database itself.  
+
+The outcome of a check on the key database is returned as an 8-bit pattern.  
+Each bit represents a specific outcome as described below:  
+
+ * 0:
+   No errors (no output)  
+
+ * Bit 1: 
+   Warning on open.  
+
+ * Bit 2:
+   Error on open.  
+
+ * Bit 3:
+   Warning on get.  
+
+ * Bit 4:
+   Error on get.  
+
+ * Bit 5:
+   Warning on set. (only checked when `-f` is used)  
+
+ * Bit 6:
+   Error on set (only checked when `-f` is used)  
+
+ * Bit 7:
+   Warning on close.  
+
+ * Bit 8:
+   Error on close.  
+
+So if the following bit pattern was returned: `010100000`  
```

Comment:
So if I mount an ini file and then remove the ini plugin and run `kdb check` I get the following output:

     ian@ian-debian-virtual:~$ sudo kdb check
     opening of kdb yield following problems: 
    4 Warnings were issued:
     Warning number: 1
	    Description: could not load module, dlopen failed
	     Ingroup: modules
	Module: dl
	At: /home/ian/Development/Elektra/iansdonnelly-libelektra/src/libloader/dl.c:80
	Reason: of module: libelektra-ini.so, because: libelektra-ini.so: cannot open shared object     file: No such file or directory
	Mountpoint: 
	Configfile: kdbOpen(): mountOpen
      Warning number: 64
	Description: could not load plugin in process plugin
	Ingroup: modules
	Module: kdb
	At: /home/ian/Development/Elektra/iansdonnelly-libelektra/src/libelektra/plugin.c:195
	Reason: ini
	Mountpoint: 
	Configfile: kdbOpen(): mountOpen
     Warning number: 13
	Description: Processing Get Plugins failed
	Ingroup: kdb
	Module: 
	At: /home/ian/Development/Elektra/iansdonnelly-libelektra/src/libelektra/backend.c:132
	Reason: elektraProcessPlugins for get failed
	Mountpoint: 
	Configfile: kdbOpen(): mountOpen
     Warning number: 65
	Description: could not reference back to plugin in process plugin
	Ingroup: modules
	Module: kdb
	At: /home/ian/Development/Elektra/iansdonnelly-libelektra/src/libelektra/plugin.c:215
	Reason: system/elektra/plugins/ini
	Mountpoint: 
	Configfile: kdbOpen(): mountOpen
    The command check failed while accessing the key database with the info:
    Error (#62) occurred!
    Description: Tried to get a key from a missing backend
    Ingroup: plugin
    Module: missing
        At: /home/ian/Development/Elektra/iansdonnelly-libelektra/src/libelektra/plugin.c:339
    Reason: system/ini_test
    Mountpoint: system/ini_test
    Configfile: